### DISCUSSION TITLE
Video Player OSD Design Idea (Updated)

### DISCUSSION THREAD

**Discussion by Perseusdehond** [jellyfin/jellyfin-vue#749] [UI & UX]
Hello there. This time I wanted to give redesigning the video player OSD a try. I don’t necessarily think that the video player OSD in 10.7.0 is bad or anything. The 10.6.x and lower ones were not that great unfortunately so a big improvement. But since this is a new client and I don’t think the current controls show the finalized design. I thought it would be good to look at a few redesigns of the player to give it an updated look. 

_I removed the old post and designs as they don't add anything with the drastic changes now made. The second design version is still there, because you can look at the comments for a great discussion with hamburlar2160 to come to this final design (special thanks to him for this design :) )_

New Design:
In this design there are quite a few changes done as suggested in this discussion thread. I think this is the most modern and clean design yet. I made the background black, because there still has to be a gradient added of some sort at the parts of the OSD.
This video player OSD has quite the resemblance to the music player OSD. This results in a more coherent design for Vue overall.
I won't go into much more detail as you can see the design versions down below :)
![JF Video Player OSD Redesign](https://user-images.githubusercontent.com/42120256/108339674-0246c880-71d8-11eb-9d98-b50d84c83d2a.png)

Old Design:
_After getting feedback on my designs, the most people liked V1 the most. As mrTimscampi suggested, I changed the icons to material design ones to keep them the same. After that I made different placements for buttons at the bottom and improved greatly upon this imo. So hear is my newer more finalized design. (The skip and reverse buttons can have different icons like the one used in JF web 10.7.0 e.g. with the addition of them showing how many seconds/minutes you skip or reverse). Thnx everyone so much for the feedback. :)**_

![Updated Player Design V1](https://user-images.githubusercontent.com/42120256/108064351-d7843500-705c-11eb-8661-85fd56f4127a.png)

Overall:
I made an addition to the time slider with darker bit at the end of the slider which represents the time when the episode ends. You could also say the time that the feat/up next popup would come. I think this would be nice to have more of a feedback and indication on this feature as a user. At last I made the skip/reverse buttons the same as in web. Although I first liked them to show the amount of time you skipped/reversed, I think showing this when you hover over them and showing it on screen when you skip/reverse would be better. 

Nothing is really that wrong with what’s in JF web in 10.7.0, but because there isn’t a finalized design for the Vue client I wanted to look at other designs. I’m not the only one, of course who uses JF and I would want the best experience for users as everyone interacts quite a lot with such an OSD. So I hope you like it ;)

**Comment by hamburglar2160** [comment]
Tidal, for their music videos, might do well here. Queue button only on the top right.
![image](https://user-images.githubusercontent.com/61879508/108002935-11066300-6fbf-11eb-99f7-6812331d71d3.png)

**Reply by Perseusdehond** [reply]
Yes, although this design is more for music playback, as there aren't titles of the content in the bottom left in JF web 10.7.0 for video playback and I think I like that. I would say that the current mini music player in Vue looks quite like this. Thanks for the suggestion
After the feedback I have changed some icons, spacing and placements of certain parts of the design. The updated design is found in the original post now.
See how you like it :)

**Reply by hamburglar2160** [reply]
I really think that there are too many settings buttons. The audio and subtitle options I really think should be combined and can be included with the gear icon.

With the titles on the bottom left, you can get some parity with the fullscreen audio player and feels a bit more modern. In the example for TV shows above it'd look like:

"Puntata 2
Season 1, Episode 2
The Grand Tour"
Though, the second line is likely unnecessary here and instead something can be used like:
"Puntata 2
The Grand Tour
Playing from: Shuffled Season 1"

"Ends At" goes top center or top left, time info goes next ot the volume button on the right.
Chromecast and Syncplay buttons on the bottom right too--they'll only show if they're enabled in Jellyfin and if a device is present on the network. You'd be looking at 4-6 buttons on the bottom right (less than/equal to what's in your top concept) and the overall OSD is less busy.

Just my two cents.

edit: I think putting the 'ends at' above the seekbar and buttons on the bottom right, with smaller text, may be even better

**Reply by Perseusdehond** [reply]
> I really think that there are too many settings buttons. The audio and subtitle options I really think should be combined and can be included with the gear icon.
>
I am not that big of a fan of putting the subtitle and audio options only in the gear icon. As it is usefull to have access to these two faster then to go in a whole menu imo. I don't really mind it if they would be combined into one button, as I did in V2 and V3 (top right), this is something that e.g. Disney Plus does.
Am I correct in assuming you also dislike the many buttons at the bottom right of YouTube. As they have even seven buttons at the bottom right?

> With the titles on the bottom left, you can get some parity with the fullscreen audio player and feels a bit more modern. In the example for TV shows above it'd look like:
>
I wouldn't necessarily want to say that it would feel more modern that way. Because an audio player is different opposed to a video player. And I don't see other services use this system except for Plex (which I really dislike the player of) and the JF 10.6 Web design (but lets don't talk about that one ;) )

> Playing from: Shuffled Season 1"
> 
I do like this idea, but it would probably show up in the menu of the queue if such a button is added. As is the case with the music player right now. I see this being a more logical place for that. But I do like that it would say something like that there for video playback too.

> "Ends At" goes top center or top left, time info goes next ot the volume button on the right.
> Chromecast and Syncplay buttons on the bottom right too--they'll only show if they're enabled in Jellyfin and if a device is present on the network. 
>
But isn't it a more logical place to put it in the top right as this is done on multiple services (except for YT web now). You also see this placement in mobile applications a lot so the current placement is more common for users

>and buttons on the bottom right, with smaller text, may be even better
>
I don't quite get what you mean. Or do you mean overall a smaller text? 
The problem I have with the placement of the endtime above the seekbar is with that it would interfere more with subtitles if those are on.

**Reply by hamburglar2160** [reply]
![image](https://user-images.githubusercontent.com/61879508/108080010-6d0fcc80-703d-11eb-9f74-40cc60e454f1.png) 
Here's what youtube TV does, which is their more updated youtube interface. The old one is hideous because it's so cluttered and hasn't really be changed in 8 years. 

Odds are the user has already selected the subs and audiotrack they want from the item details screen, so does it really need to be plastered all over the OSD?

**Reply by Perseusdehond** [reply]
> Odds are the user has already selected the subs and audiotrack they want from the item details screen, so does it really need to be plastered all over the OSD?
>
I do agree with you on that. What do you think of the 10.7 web OSD design?

**Reply by hamburglar2160** [reply]
> > Odds are the user has already selected the subs and audiotrack they want from the item details screen, so does it really need to be plastered all over the OSD?
> 
> I do agree with you on that. What do you think of the 10.7 web OSD design?

Good, but antiquated

**Reply by Perseusdehond** [reply]
> Good, but antiquated
>
Do you have a concept/design you really like which is in your eyes modern and would fit the Vue video player? Screenshot/link if possible :)

**Reply by hamburglar2160** [reply]
https://listen.tidal.com/view/pages/videos screenshotted above. go full screen here. also have a look at the fullscreen music player implemented by ferferga.

**Reply by Perseusdehond** [reply]
Hmm, I do really like the Tidal OSD (never used it before). As it feels really clean, but there is still a problem with the design. That it is designed for music (video's) playback. 

I do really like the idea of such a design for the video player OSD in Vue though, which would be more inline with the music player OSD by ferferga as he did a wonderful job with that.

The shuffle and repeat buttons can be replaced with the skip and rewind buttons, so there are not more than 5 buttons at the middle. I do like the addition of queue added to the OSD which would bring up a menu at the side (like in tidal/a bit like the current queue for the music player) especially for episodes and such.

I would suggest keeping the time at both the left and right as is in the music OSD (Vue) instead of underneath the time slider at the right, because there would be more buttons for playback.

The heart button would go away and I could move the cast button to the bottom right.

Question, what would you do with the syncplay button/feature in the Jellyfin OSD, as I don't think you want to keep it top right?

Then I will add the name of the movie/episode bottom left. I guess you would put a back button in the top left? And where would you put the picture-in-picture button? As I don't think there is a way to combine the full screen and pip button in a logical way (not that you suggested that to happen).

You also wanted the audio and subtitle button to be combined into one instead of 2 buttons or for it to be completely moved to a menu. Although this menu would require a proper design as I don't want users to have to click too many times to activate one feature.

I will try and fiddle it with it a bit more to achieve a more modern look and I'll let you know :)

**Reply by Perseusdehond** [reply]
I also thought of that users maybe could pin certain buttons to the OSD in the playback settings, so options in the settings menu (in the video player OSD) could become their own icon. For example when you don't want to have the subtitle button on the OSD, you could simply unpin this and it would appear in the settings menu in the video player OSD. 
This would allow for more customization by the user which I'm a fan of.

**Reply by hamburglar2160** [reply]
> Question, what would you do with the syncplay button/feature in the Jellyfin OSD, as I don't think you want to keep it top right?

I can't speak for everybody but I don't have syncplay enabled on my server so the button doesn't show at all. Likewise, the Cast icon wouldn't appear unless a Cast device is detected on the network. I would personally tack them on to the bottom right because that's where all the other buttons are but /shrug

> Then I will add the name of the movie/episode bottom left. I guess you would put a back button in the top left? And where would you put the picture-in-picture button? As I don't think there is a way to combine the full screen and pip button in a logical way (not that you suggested that to happen).

I don't have a strong opinion. To keep things consistent the back button should be on the top left. PIP across Jellyfin clients (and streaming services?) that support it is on the bottom right so why change that?

> You also wanted the audio and subtitle button to be combined into one instead of 2 buttons or for it to be completely moved to a menu. Although this menu would require a proper design as I don't want users to have to click too many times to activate one feature.

They are settings after all and there is a settings button sitting right there! The subtitle and audio tracks have been determined by default or by the user before loading the video.

**Reply by Perseusdehond** [reply]
Yeah, I am liking your proposal more and more. It is very late for me so I just made a quick first version of your idea/tidal player. Let me know if I'm on the right track for this ;)

![Late Night Player Example ](https://user-images.githubusercontent.com/42120256/108132981-1a252c00-70b4-11eb-9c3e-2fe4e20bee9f.png)

**Reply by hamburglar2160** [reply]
I don't know what I think of the current/end times at either end of the seekbar, but I really like this.

**Reply by Perseusdehond** [reply]
Yeah I do really like it too. I moved the time to beneath the title and made the text smaller. I changed the skip/reverse buttons. I also made the back button darker as this is currently the only button up top, so I didn't want it sticking out as much.

As I do think see quite some comparisons to Plex with some other button placements/additions. I was asking myself what you thought of their design and if you wanted to see some things added/moved to different places (although I don't like the big bars on the top and bottom)
Updated design:
![Updated V1 Design 2](https://user-images.githubusercontent.com/42120256/108179854-60a37680-7106-11eb-9a63-03e7a3bfd71d.png)
Plex design:
![Plex OSD](https://user-images.githubusercontent.com/42120256/108179973-84ff5300-7106-11eb-9b83-c789fbcb3f37.PNG)

**Reply by Perseusdehond** [reply]
I also thought that my buttons were a bit big in comparison to the current music player OSD sizes, which would cause the OSD to feel more cluttered than necessary. This is why I went ahead and changed the sizes of all the buttons to the seize that they have in the music player. This also reduces the thickness of the time slider. This feels way better again. I'm not too happy with the bottom left though, so maybe the time has to be next to the title? Let me know what you think ;)

![Updated Player Design 3](https://user-images.githubusercontent.com/42120256/108197937-47f18b80-711b-11eb-842e-dc2e32429b6c.png)

Edit:
I could also do something like this. So I removed the time beneath the title and put it underneath the play controls as this is nice and centralist. Then I added the 'Ends at' at the left next to the title, because I felt it was a little bit too empty compared to the right.
I made the background black so the OSD is better shown of as there needs to be some contrast against the background still imo.

![Updated Player Design 4](https://user-images.githubusercontent.com/42120256/108200868-55a91000-711f-11eb-80fd-d9f8afb3a475.png)

**Reply by hamburglar2160** [reply]
![image](https://user-images.githubusercontent.com/61879508/108204227-8a9e6e00-70f1-11eb-8e62-0c17af5da500.png)
For a desktop view, no need to abbreviate "S2:E1," so "Season 2, Episode 1." That can likely have its own line.

The volume slider, to my eye, looks funny sandwiched after the settings gear. I wonder if a vertical slider would interfere with the seekbar too much.

I haven't tried the Fullscreen Video PR yet so will playback default to full screen or PIP as is current? Will the arrow on the top left ultimately be necessary? Maybe a stop button will be called for, especially depending on what happens with the player when somebody initiates a cast session from this view.

**Reply by Perseusdehond** [reply]
So with these changes it would look like this:
![Updated Player Design 5](https://user-images.githubusercontent.com/42120256/108209674-dcafb580-712a-11eb-8a9d-2761d442eef0.png)

With the volume slider changed to a vertical one it would look like this:
![Updated Player Design 5 Variation](https://user-images.githubusercontent.com/42120256/108210289-9f97f300-712b-11eb-865b-568f1379152e.png)

**Reply by Perseusdehond** [reply]
> I haven't tried the Fullscreen Video PR yet so will playback default to full screen or PIP as is current? Will the arrow on the top left ultimately be necessary? Maybe a stop button will be called for, especially depending on what happens with the player when somebody initiates a cast session from this view.
>
I don't know what it's behaviour will be. But I thought it would only make the full screen button work. So I think it would open in pip as is current.
The arrow in the top left is for closing the player. Maybe an arrow isn't the best icon, but it is the same as in current web. If you look at plex they moved it in the bottom center next to the  controls.

**Reply by hamburglar2160** [reply]
I should have been more clear about the vertical slider. It'd appear above the volume button on hover. If you are adamant about keeping the volume slider in the middle of the button collection I think it's worth considering, otherwise keep it expanded horizontally between the time info and syncplay button. I like this proposal overall though, I wonder where others would take it.

I'm not sure I'd take any cues from Plex here...

**Reply by Perseusdehond** [reply]
I don't really like the Plex OSD myself, but as you weren't sure about the arrow button top left you could do something like them.

I personally think it would interfere too much with the time slider and it would look/feel good to use. So keeping it horizontally is better imo then. I think this is the best way it has looked. The only thing is that the settings menu will obstruct the Ends at time, but that is no big deal. 
![Updated Player Design 7](https://user-images.githubusercontent.com/42120256/108215643-d6710780-7131-11eb-84de-2a7616907800.png)

**Reply by Perseusdehond** [reply]
This is the best version for a new design I think. Only need to make it stick out on for example very bright images/video's

Would you change/add anything more?

**Reply by hamburglar2160** [reply]
I really like it. What would the bottom left look like for a movie? For both cases I don't know if I'd use larger text for the title but they look good.

**Reply by Perseusdehond** [reply]
Well the larger the text, the more the time slider has to go up, so I didn't want to make it too big. The times are also not that big either. Will make a version for a movie

**Reply by Perseusdehond** [reply]
I think the text for a movie can be made bigger. I now took a larger title and I think this looks really nice

![Updated Player Design Movies](https://user-images.githubusercontent.com/42120256/108216975-564ba180-7133-11eb-998c-550a195530bc.png)

**Reply by hamburglar2160** [reply]
For my tastes I'd stick with the same text sizes used by the music player. I think it's distracting. What happens when the movie title is "Birds of Prey: and the Fantabulous Emancipation of one Harley Quinn?"

**Reply by hamburglar2160** [reply]
I don't know if this has any place in Jellyfin, but one more cue from Tidal:
![image](https://user-images.githubusercontent.com/61879508/108218034-41a2e580-7102-11eb-8b0b-e9518c7de195.png)

**Reply by Perseusdehond** [reply]
> I don't know if this has any place in Jellyfin, but one more cue from Tidal:
> ![image](https://user-images.githubusercontent.com/61879508/108218034-41a2e580-7102-11eb-8b0b-e9518c7de195.png)

I like that, only thing being that we don't currently have that down there. If you press on the mini album picture, you would automatically go into full screen. Because that bar is the 'pip'/miniplayer design for the music player.
For video players, we don't currently have such a bar at the bottom to have control over the movie in pip mode. I do like the idea though when we have such a design.

**Reply by Perseusdehond** [reply]
> For my tastes I'd stick with the same text sizes used by the music player. I think it's distracting. What happens when the movie title is "Birds of Prey: and the Fantabulous Emancipation of one Harley Quinn?"

I agree, and this happens with the small text size (same as music player)
![Updated Player Design Movies](https://user-images.githubusercontent.com/42120256/108219571-2782fa80-7136-11eb-959d-4e93014a7199.png)

**Reply by Perseusdehond** [reply]
So I made a settings menu for this design as there has to be a proper better one than there is in web now. So my idea was to keep the current selectors for subtitles and audio etc.
Then I added some sliders and tried to group the selector type together. Don't know if I missed any big ones, but I think this looks quite nice. Let me know what you think :)

![Settings Menu Design 1](https://user-images.githubusercontent.com/42120256/108241019-b0586100-714b-11eb-8b32-e1a26920bf1b.png)

**Reply by hamburglar2160** [reply]
I don't like pulldown menus here or the use of _both_ sliders and pulldown menus.
There will likely be a Jellyfin subtitles settings view, so linking to that would be useful. Youtube TV does this:
![image](https://user-images.githubusercontent.com/61879508/108246810-dd8f1a00-711f-11eb-9255-d00430cf4dd6.png)
I am not sure I see the point of having either the autoplay toggle or the aspect ratio settings. The transcoding can be simpler too.
![image](https://user-images.githubusercontent.com/61879508/108247119-3c549380-7120-11eb-8f58-36ed49fdc846.png)
"Quality" instead of "Video," "Audio Track" if alternatives exist instead of "Audio," and I'd do away with the icons since they're redundant.

**Reply by Perseusdehond** [reply]
Thanks for the feedback. I took the aspect ratio and autoplay settings from the settings menu in web. As you can fullscreen it for example. And the autoplay just tells you if you want to autoplay the next episode/movie so I guess you would want it to be there. I would be okay with dropping the aspect ratio setting as I don't feel there is much use, but maybe some users do like and use this feature.

Btw, I like pull down menu's here I guess, because the menu size will stay the same and it is also used on the movie items page. This also allows for faster editing settings instead of having to open the settings page everytime again.

I simplified the rest and removed the sliders in exchange for drop down menu's.
Is this better?

![Settings Menu 2](https://user-images.githubusercontent.com/42120256/108251238-6e351c80-7157-11eb-8c75-3347c4f4eec1.png)

**Reply by hamburglar2160** [reply]
What happens if there's like 20 subtitle options or more than 5 transcoding qualities available (I certainly hope that there aren't).

**Reply by Perseusdehond** [reply]
It would get a scrollbar as is the case in Vue now

![image](https://user-images.githubusercontent.com/42120256/108252169-9cffc280-7158-11eb-8440-96f39d8fda9b.png)

**Reply by Perseusdehond** [reply]
For me there are more transcoding quality's available in Web. It isn't just the Youtube 1080p, 720p etc., but also the amount of Mbps which results in quite a few options more

**Reply by Perseusdehond** [reply]
Another question: do I need to add a queue icon bottom right when playing an episode of a series? As this would be more like the music player and you can rearrange the episodes etc./when you are playing from a playlist

**Reply by hamburglar2160** [reply]
> For me there are more transcoding quality's available in Web. It isn't just the Youtube 1080p, 720p etc., but also the amount of Mbps which results in quite a few options more

Most of which has already been predetermined in the user settings or the admin dashboard or the item details screen.

**Reply by Perseusdehond** [reply]
Well on the item details screen, you can't select the quality. That's actually strange when you think about it.
And I haven't seen an option in user settings/admin dashboard to show just 1080p, 720p etc.

**Reply by ferferga** [reply]
In my opinion in doesn't make any sense to give any quality selector (besides the version one, which I think it's the one @hamburglar2160 is confusing) because you won't know nothing about the transcoded quality (as it will depend on network speed, etc etc) once you start playing a video.

**Reply by Perseusdehond** [reply]
@ferferga I think you mean to not show the quality selector on the item page right? Because I'm alright with that. Does Jellyfin automatically transcode or can you also play the original file (when your speed is high enough)? Because in the current web it isn't clear for me. E.g if I set it to automatic, it still says transcoding - 60 mbps while the file clearly doesn't need 60 mbps to be played/neither that it needs transcoding. 
Sorry if this is a dumb question :)

**Reply by ferferga** [reply]
@Perseusdehond Yes of course

This is my proposal for how a real quality switcher should work in the future: https://github.com/jellyfin/jellyfin/discussions/4795

Right now everythig's hardcoded and there's no real "intelligence" behind the transcoding process, all works thanks to a set of hardcoded stuff.

Automatic should play the original file, if it transcodes it's because it thinks your file is not compatible based on that hardcoded stuff. It plays with the original bitrate of the file iirc, but I don't know enough of how server works to assure this.

**Reply by Perseusdehond** [reply]
> @Perseusdehond Yes of course
> 
I agree on that then.

> This is my proposal for how a real quality switcher should work in the future: [jellyfin/jellyfin#4795](https://github.com/jellyfin/jellyfin/discussions/4795)
> 
I really like this proposal. Way easier for users to just play in the best quality and don't worry about anything else

> Automatic should play the original file, if it transcodes it's because it thinks your file is not compatible based on that hardcoded stuff. It plays with the original bitrate of the file iirc, but I don't know enough of how server works to assure this.
>
Ah okay, thnx, it only isn't always clear what it is doing (because of the bitrate and transcoding text).
